survey-sdk/src/main/java/co/nimblehq/survey/sdk/api/AppService.kt

```diff
@@ -0,0 +1,25 @@
+package co.nimblehq.survey.sdk.api
+
+import co.nimblehq.survey.sdk.entity.SurveyEntity
+import moe.banana.jsonapi2.ArrayDocument
+import moe.banana.jsonapi2.ObjectDocument
+import retrofit2.http.GET
+import retrofit2.http.Path
+import retrofit2.http.Query
+
+interface AppService {
+
+    @GET("/api/{version}/surveys")
+    suspend fun getSurveyList(
+        @Query("page[number]") page: Int,
+        @Query("page[size]") size: Int,
+        @Path("version") version: String = "v1",
+    ): ArrayDocument<SurveyEntity>
+
+    @GET("/api/{version}/surveys/{id}")
+    suspend fun getSurveyDetail(
+        @Path("id") id: String,
+        @Path("version") version: String = "v1",
+    ): ObjectDocument<SurveyEntity>
```

Comment:
Wait... I don't think the response from ApiServie should be wrapped by `ArrayDocument` or `ObjectDocument`. 

getSurveyList -> `Response<List<SurveyEntity>>`
getSurveyDetail -> `Response<SurveyEntity>`

cc @luongvo What do you think?

Reply:
@hoangnguyen92dn Could you please let me know why we should wrap the result with `Response`? 
I think it belongs to the mapping data.

Reply:
@doannimble The first thing is Service return the `Response` then you need to validate the response is success or not before getting the body 🤷‍♂️  I think we need to have a `Repository` layer to handle mapping response to model rather than returning the `SurveyEntity`, this entity is using for handle api response purpose and keep it inside the SDK instead.
And the second thing is you are returning `ArrayDocument` with `ObjectDocument` to the `SurveyApi`, if I understand right, in the integration step, the sample app will call the `SurveyApi` to get the list of Survey and SurveyDetails, then why does the sample app need to handle `ArrayDocument` and `ObjectDocument`?

Reply:
@hoangnguyen92dn 
I agree with your point. Perhaps, we need to make our SDK more convenience by return the Result which corresponding entity.
But to make it more simple, I will not define the `Repository`, just wrap up everything in `Result` class. Is it good enough?

Reply:
@doannimble I want to clarify here 🤔  What is the `Result` here, Result from Kotlin or customized one and the purpose of `Result` class?

Reply:
@hoangnguyen92dn Exactly is the `ResultSdk` - which define the data response is `Success` or `Error`. This class is generated from AS template and I renamed it to avoid confusing.
Please take a look on my update 🙏

Reply:
@doannimble `Result` is still fine then 😁

Reply:
- [Response<T>](https://square.github.io/retrofit/2.x/retrofit/retrofit2/Response.html): is a specific interface type that we can instruct Retrofit to returns more details other than the mapped data object <T> we defined, it means the response will include `body()` or `code()` or `success()` that you can manipulate the result afterward, otherwise, if we define returns `<T>` then retrofit will try to map the `body()` automatically for us and throws the Exception on error cases.
- `Result` or `ResultSdk` we are talking about, is the communication bridge between the SDK and the main-app, it can be `Failed` or `Success`, and the main app doesn't need to know about the `Response<T>` class, which is the dependency on the Retrofit.

When designing an SDK, you will need to put yourself in the shoes of both the users and the sdk designer. As a programmer, do you want to use an SDK that requires you to step in the parsing steps to get the data you wanted? If not, then we need to make sure, as an sdk designer, you must returns only what the users (dev) wanted.